File: app/src/main/java/org/neidhardt/dynamicsoundboard/mediaplayer/MediaPlayerComparator.java
package org.neidhardt.dynamicsoundboard.mediaplayer;

import android.support.annotation.NonNull;
import org.neidhardt.dynamicsoundboard.dao.MediaPlayerData;

import java.util.Comparator;

/**
 * File created by eric.neidhardt on 19.06.2015.
 */
public class MediaPlayerComparator implements Comparator<MediaPlayerData>
{
	@Override
	public int compare(@NonNull MediaPlayerData lhs, @NonNull MediaPlayerData rhs)
	{
		if (lhs.getSortOrder() == null)
			return -1;
		if (rhs.getSortOrder() == null)
			return 1;

		return lhs.getSortOrder().compareTo(rhs.getSortOrder());
	}
}

File: app/src/main/kotlin/org/neidhardt/dynamicsoundboard/mediaplayer/MediaPlayerComparator.kt
package org.neidhardt.dynamicsoundboard.mediaplayer

import org.neidhardt.dynamicsoundboard.dao.MediaPlayerData

import java.util.Comparator

/**
 * File created by eric.neidhardt on 19.06.2015.
 */
class MediaPlayerComparator : Comparator<MediaPlayerData>
{
	override fun compare(lhs: MediaPlayerData, rhs: MediaPlayerData): Int
	{
		if (lhs.sortOrder == null)
			return -1
		if (rhs.sortOrder == null)
			return 1

		return lhs.sortOrder!!.compareTo(rhs.sortOrder)
	}
}

